docs/ds/cartesian-tree.md

```diff
@@ -1,71 +1,71 @@
 author: sshwy, zhouyuyang2002, StudyingFather, Ir1d, ouuan, Enter-tainer
 
-本文介绍一种不太常用，但是与大家熟知的平衡树与堆密切相关的数据结构——笛卡尔树。
+This article introduces the [Cartesian tree](https://en.wikipedia.org/wiki/Cartesian_tree) - a data structure that is not very commonly used, but is closely related to the well-known balanced tree and heap.
 
-笛卡尔树是一种二叉树，每一个结点由一个键值二元组 $(k,w)$ 构成。要求 $k$ 满足二叉搜索树的性质，而 $w$ 满足堆的性质。一个有趣的事实是，如果笛卡尔树的 $k,w$ 键值确定，且 $k$ 互不相同， $w$ 互不相同，那么这个笛卡尔树的结构是唯一的。上图：
+Cartesian tree is a type of binary tree, in which each node consists of a key value tuple $(k,w)$ . It is required that $k$ satisfy the properties of a binary search tree, and $w$ satisfy the properties of a heap. An interesting fact is that if the $k,w$ key value of the Cartesian tree is already determined, and both $k$ and $w$ are different from each other, then the structure of the Cartesian tree is unique. Please see the figure below：
 
 ![eg](./images/cartesian-tree1.png)
 
-（图源自维基百科）
+(Figure from Wikipedia)
 
-上面这棵笛卡尔树相当于把数组元素值当作键值 $w$ ，而把数组下标当作键值 $k$ 。显然可以发现，这棵树的键值 $k$ 满足二叉搜索树的性质，而键值 $w$ 满足小根堆的性质。
+The Cartesian tree above is equivalent to taking the array element value as the value $w$ and the array index as the key $k$ . Obviously, it can be found that the key of this tree $k$ satisfies the property of the binary search tree, and the value $w$ satisfies the property of min heap.
```

Comment:
```suggestion
The Cartesian tree above is equivalent to taking the array element value as the value $w$ and the array index as the key $k$ . Obviously, it can be found that the key of this tree, $k$, satisfies the property of the binary search tree, and the value $w$ satisfies the property of min heap.
```
不然感觉 k 是个树了